### PR TITLE
Enables the empty_count SwiftLint rule

### PR DESCRIPTION
**Pull request by qsmmorgan**
Replaces most instances of `array[0]` with `.first`

### CODE REVIEW

File: arcgis-ios-sdk-samples/Maps/Create and save a map/CreateOptionsViewController.swift [hunk 1]
```diff
@@ -143,13 +143,11 @@ class CreateOptionsViewController: UIViewController, UITableViewDataSource, UITa
         let basemap = self.basemaps[self.selectedBasemapIndex].copy() as! AGSBasemap
         
         //create an array of the selected operational layers
-        var layers = [AGSLayer]()
-        for index in self.selectedLayersIndex {
-            let layer = self.layers[index].copy() as! AGSLayer
-            layers.append(layer)
+        let selectedLayers: [AGSLayer]? = selectedLayersIndex.isEmpty ? nil : selectedLayersIndex.map { (index) -> AGSLayer in
```
Comment (philium) [inline arcgis-ios-sdk-samples/Maps/Create and save a map/CreateOptionsViewController.swift]: This, because of the ternary operator, is a little hard to follow. Maybe something like this would be more readable?

```swift
let selectedLayers: [AGSLayer]?
if !selectedLayersIndex.isEmpty {
    selectedLayers = selectedLayersIndex.map { layers[$0].copy() as! AGSLayer }
} else {
    selectedLayers = nil
}
```

File: arcgis-ios-sdk-samples/Search/Offline geocode/OfflineGeocodeViewController.swift [hunk 2]
```diff
@@ -148,7 +153,8 @@ class GeocodeOfflineViewController: UIViewController, AGSGeoViewTouchDelegate, U
                 //if a result is found extract the required attributes
                 //assign the attributes to the graphic
                 //and show the callout
-                if let results = results, results.count > 0 {
+                if let results = results,
```
Comment (philium) [inline arcgis-ios-sdk-samples/Search/Offline geocode/OfflineGeocodeViewController.swift]: Should this be `if let result = results.first { … }`?

File: arcgis-ios-sdk-samples/Search/Reverse geocode/ReverseGeocodeViewController.swift [hunk 3]
```diff
@@ -79,7 +79,8 @@ class ReverseGeocodeViewController: UIViewController, AGSGeoViewTouchDelegate {
                 }
             }
             else {
-                if let results = results, results.count > 0 {
+                if let results = results,
```
Comment (philium) [inline arcgis-ios-sdk-samples/Search/Reverse geocode/ReverseGeocodeViewController.swift]: It seems like this should be `if let result = results.first { … }`.

File: arcgis-ios-sdk-samples/Maps/Create and save a map/CreateOptionsViewController.swift (hunk 1 above)
Comment (qsmmorgan) [reply arcgis-ios-sdk-samples/Maps/Create and save a map/CreateOptionsViewController.swift]: Looks good, thanks
